docs/master/api-reference/directives.md

```diff
@@ -2928,31 +2945,67 @@ that gets actors over age 37 who either have red hair or are at least 150cm.
 }
 ```
 
-Lighthouse generates definitions for an `Enum` type and an `Input` type
-that are restricted to the defined columns.
+By default, the `@whereConstraints` directive allows for filtering on all fields. To restrict the allowed fields, you can pass in an array of field names to the `columns` parameter:
 
 ```graphql
-"Dynamic WHERE constraints for the `where` argument on the query `people`.
-input PeopleWhereWhereConstraints {
+type Query {
+    people(
+        where: WhereConstraints @whereConstraints(columns: ["age", "height"])
+    ): [Person!]! @all
+}
+```
+
+When using the default `WhereConstraints` input, Lighthouse allows for filtering on any of the fields in the model, specified as strings in the request. To customize this behavior, you can define your own input and enum:
+
+```graphql
+"Dynamic WHERE constraints for the `where` argument on the query `people`."
+input PeopleWhereConstraints {
```

Comment:
That should not be necessary. Simply specifying the column names in `@whereConstraints` will manipulate the original argument definition to point towards a generated type that holds the correct enum for the allowed columns.

Reply:
I just tested and I think it's working as you intend:
- If you do not specify the `columns` argument **and** you use the built-in input, columns **must** be queried by string.
- If you do specify the columns **or** you use a custom input like the examples show, columns **must** be queried by the enums.

Ultimately, my PR hopes to clear up some of these fiddly pitfalls that burned so much of my time trying to get this directive working. I think this functionality should be explicitly spelled out in the docs.

Reply:
Using a custom input is not necessary, but i see how the docs could give the wrong impression. Maybe we can hint that the shown code is just autogenerated and not meant to be included in the schema.